main.js

```diff
@@ -92,41 +92,49 @@ function removeError(){
-
-    const form = div_form_container.appendChild(document.createElement("form"));
-    form.setAttribute("class", "form");
-    form.setAttribute("id", "login")
-
-    const title = form.appendChild(document.createElement("h1"));
-    title.textContent = "Login form";
+    createFormContainer();
 
-    const div_form_userName = form.appendChild(document.createElement("div"));
-    div_form_userName.setAttribute("class", "form_input");
-    div_form_userName.setAttribute("id", "form_username");
-    div_form_userName.innerHTML = '<label for="username">Username:</label>' +
-    '<input type="text" name="username" id="username" placeholder="username">';
+    const parentElement = document.getElementById("login");
+    createTitle(parentElement, "Login form");
+    createInput(parentElement);
+    createButton(parentElement); 
+}
 
-    const div_form_password = form.appendChild(document.createElement("div"));
-    div_form_password.setAttribute("class", "form_input");
-    div_form_password.setAttribute("id", "form_password");
-    div_form_password.innerHTML = '<label for="password">Password:</label>' +
-    '<input type="password" name="password" id="password" placeholder="password">';
+function createFormContainer(){
+    const div_form_container = document.createElement("div");
+    div_form_container.setAttribute("class", "form_container");
 
-    const div_button = form.appendChild(document.createElement("div"));
-    div_button.setAttribute("class", "button");
-    div_button.innerHTML = '<button id="login_btn">login</button>';
-    
     document.body.append(div_form_container);
-    return div_form_container;
-}
-
 
+    reusebleElements({parentElement: div_form_container, element: "form", class: "form", id: "login"});
+}
 
+function createTitle(parentElement, text){
+    const title = parentElement.appendChild(document.createElement("h1"));
+    title.textContent = text;
+}
 
+function createInput(parentElement){
+    reusebleElements({parentElement: parentElement, element: "div", class: "form_input", id: "form_username"});
+    form_username.innerHTML = '<label for="username">Username:</label>' +
+    '<input type="text" name="username" id="username" placeholder="username">';
 
+    reusebleElements({parentElement: parentElement, element: "div", class: "form_input", id: "form_password"})
+    form_password.innerHTML = '<label for="password">Password:</label>' +
+    '<input type="password" name="password" id="password" placeholder="password">';
+}
 
+function createButton(parentElement){
+    reusebleElements({parentElement: parentElement, element: "div", class: "button", id: ""})
+    document.querySelector(".button").innerHTML = '<button id="login_btn">login</button>';
+}
 
+//
+//{parentElement: "parent element", element: "element you want to create", id: "idName", class: "idClass" }
+function reusebleElements(object){
+    const new_element = object.parentElement.appendChild(document.createElement(object.element));
```

Comment:
Definition:
A pure function is the function that get some data from the params and return new value depends of params. Pure function cannot create any of side effects in your application.

Example of side effects:

- `object.parentElement.appendChild` call. This is a side effect because it change outer environment of the function (parentElement).
- `console.log` call. This is a side effect because it also change outer environment of the function - your developer console.
- alert() call. This is a side effect..

Let's make this function _pure_. 


more...

**Example of pure function**:
```
const a = 10;
const b = 12;

function add(firstNumber, secondNumber) {
   return firstNumber + secondNumber;
}

const result = add(a, b);
// a = 10, b = 12, result = 22
```

**Example of function with side effects**:
```
const a = 10;
const b = 12;
let result;

function add(firstNumber, secondNumber) {
   result = firstNumber + secondNumber;
}

add(a, b);

// a = 10, b = 12, result = 22
```

Reply:
Done